### PR TITLE
Move build from circleCI to drone

### PR DESCRIPTION
**Pull request by captncraig**
This moves all builds from circleCI to drone. On pull request does all lint and build steps.

On push to master it will do all that, as well as publishing docker image and packages.

### CODE REVIEW

**Comment by captncraig** [comment]
I tested the docker and package publishing. All credentials seem to be working.

We should probably disable the circleCI jobs before merging this.

File: .drone.jsonnet [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      './scripts/version',
+      './scripts/version > .tags', // save version in special file for docker plugin
+      'make build',
+    ]),
+    // We can't use 'make docker' without making this repo priveleged in drone
+    // so we will use the native docker plugin instead for security.
+    step('docker build',[],'plugins/docker')+{
+      settings:{
+        repo: repo,
+        dry_run: 'false',
```
Comment (captncraig) [inline .drone.jsonnet:52]: @mem my intention was that the docker build step could run on all prs and things, and the docker push step would only run on master. By setting dry_run:false here, you will be pushing to docker hub on each pr build and stuff. Is that what you intended?

Comment (mem) [reply .drone.jsonnet]: no, I misread the config and thought this section applied to master. Reverted.

But it did fail and I was expecting it not to. Am I missing something else?

Comment (mem) [reply .drone.jsonnet]: I found what I was missing, sorry about the noise!

**Pull request merged**